viral/fix_tiff.py

```diff
@@ -0,0 +1,132 @@
+"""
+Utils to fix broken tiff files by saving all but the last, presumably broken, frame.
+Recommended to fix in this order:
+1. If you set up suite2p to be verbose (print the tiff file name when loading), you can skip step 2.
+2. Use "find_broken_tiffs_in_session" to get a list of broken tiff files in a given session directory.
+3. Use "save_tiff_until_broken" on each broken tiff file to save a new tiff file with all but the last frame.
+4. Carefully move the broken file from the original directory to the "damaged_tiffs" directory.
+5. Now re-run suite2p on the session directory.
+"""
+
+import tifffile
+import sys
+import re
+import numpy as np
+from pathlib import Path
+from typing import List, Tuple
+from ScanImageTiffReader import ScanImageTiffReader
+from natsort import natsorted
+
+HERE = Path(__file__).parent
+sys.path.append(str(HERE.parent))
+sys.path.append(str(HERE.parent.parent))
+
+from viral.constants import TEMP_CACHE_PATH, TIFF_UMBRELLA
+from viral.cache_2p_sessions import extract_metadata, check_no_dropped_frames
+
+
+def extract_metadata_broken_tiff(
+    tiff_path: Path,
+) -> Tuple[int, np.ndarray, List[float]]:
+    """Extract metadata from tiff file until the broken frame (last frame)"""
+    with tifffile.TiffFile(tiff_path) as tiff:
+        n_frames = len(tiff.pages)
+        timestamps = list()
+        for idx in range(n_frames - 1):
+            # read all but the last, presumably broken, frame
+            description = tiff.pages[idx].tags["ImageDescription"].value
+            timestamp_match = re.search(
+                r"frameTimestamps_sec\s*=\s*(-?\d+\.\d+)", description
+            )
+            timestamps.append(float(timestamp_match[1]))
+
+        if timestamps is None or len(timestamps) != n_frames - 1:
```

Comment:
```suggestion
        if any([t is None for t in timestamps])
```
timestamps will never be None as you initialise it as a list and the list length will always be n_frames - 1 as you will append None to the list if there's no match